Update spell coordination schedule to the post-goerli-world

SidestreamColdMelon opened the pull request:
Closes https://github.com/makerdao/pe-checklists/issues/23

Since we plan to deprecate goerli testnet after February 21, 2024, we also need to update our coordination schedule and remove outdated information. This PR:
- Removes goerli from the schedule (since it's being deprecated)
- Removes old cadence proposal linked in current checklists (since it becomes irrelevant)
- Introduces basic table format instead of a gantt chart or a list
- Introduces one-row-per-role structure to better outline sequence of events for the outsiders
- Splits crafters checklists into relevant sections (to match stages described in the schedule)
- Updates [currently used schedule](https://github.com/makerdao/pe-checklists/issues/13)
  - Crafting still starts on Friday
  - Most of the deadlines are moved to 16:00 UTC instead of 23:59 CET
  - New "sync" session is proposed in the end of the spell to streamline handover process

amusingaxl commented on spell/spell-crafter-mainnet-workflow.md:
```diff
@@ -4,11 +4,35 @@
+| Reviewers   | Spell code is reviewed (against the Exec Doc)      | 16:00 UTC Week 2 Tuesday         |
+| Crafter     | Spell is deployed, Testnet is created              | 12:00 UTC Week 2 Wednesday       |
+| Reviewers   | Spell deployment is approved                       | 16:00 UTC Week 2 Wednesday       |
+| Crafter     | Handover is published                              | 16:00-16:30 UTC Week 2 Wednesday |
```
For clarity, I believe this could be phrased as:
```suggestion
| Crafter     | Spell address is published                              | 16:00-16:30 UTC Week 2 Wednesday |
```

Same for the items below.

Other than that, changes look good.

SidestreamColdMelon replied:
Thanks, updated the wording

SidestreamColdMelon commented:
@amusingaxl as you're fine with the proposal to have "two reviews" – one for the code based on the sheet, and one to validate addressed review + the exec doc, I would rearrange the checklist items to account for this process change. For example checks relevant to exec doc can go to the bottom of the "Development Stage" or even moved into a separate new section between "Development" and "Deployment", eg "Pre-Deployment". This way, the reviewers wouldn't have to jump around the checklists and hold publishing them till the very end

amusingaxl reviewed (approved):
LGTM

VoteWizard reviewed (approved):
LGTM

SidestreamStrongStrawberry reviewed (approved):
lgtm :)
